docs-src/how-to-create-layoutbuilder.md

```diff
@@ -14,10 +14,303 @@ kernelspec:
 How to create arrays with LayoutBuilder (more control)
 ======================================================
 
-**This is a stub:** I intend to write this article, but haven't yet.
+What is LayoutBuilder
+---------------------
 
-If you need it soon, create an issue saying so and I'll make it a higher priority.
+`ak.layout.LayoutBuilder` is the low-level LayoutBuilder that builds layouts, or `ak.layout.Content` arrays. It must be initialized by a JSON string that represents a valid `ak.forms.Form`.
 
-[![](img/github-issues-documentation.png)](https://github.com/scikit-hep/awkward-1.0/issues/new?assignees=&labels=docs&template=documentation.md&title=)
+A layout is the composable `ak.layout.Content` elements that determine how an array is structured. The layout may be considered a "low-level" view, as it distinguishes between arrays that have the same logical meaning (i.e. same JSON output and high-level `type`) but different
+  * node types, such as `ak.layout.ListArray64` and `ak.layout.ListOffsetArray64`,
+  * integer type specialization, such as `ak.layout.ListArray64` and `ak.layout.ListArray32`,
+  * or specific values, such as gaps in a `ak.layout.ListArray64`.
 
-The text of your issue doesn't have to be much more than a link to this page, so I can be sure which page you're referring to. If you add details about how and why you need it, however, I may be able to tailor the text to help you more.
+`ak.forms.Form` describes a low-level data type or "form". There is an exact one-to-one relationship between each `ak.layout.Content` class and each Form.
```

Comment:
Maybe lowercase form, or link to `ak.forms.Form` again?
```suggestion
`ak.forms.Form` describes a low-level data type or "form". There is an exact one-to-one relationship between each `ak.layout.Content` class and each form.
```

Reply:
I'd prefer capitals on "Form" because it's such a common word and easily mis-seen or even mistyped as "from." I try to use capitals with "Form" because it clues people in to "this is some special thing with a specific meaning."

All of these references to classes could point to the documentation: instead of "Form" or even "`ak.forms.Form`", you could say [ak.forms.Form](https://awkward-array.readthedocs.io/en/latest/ak.forms.Form.html). Similarly with their subclasses, including Content subclasses like [ak.layout.ListArray](https://awkward-array.readthedocs.io/en/latest/ak.layout.ListArray.html). I think the other tutorials do this.